feature: add Jupyter Server and supervisord config to template

zuoyuanh opened the pull request:
*Description of changes:*

* Add default config for Jupyter server
* Add script to start Jupyter server
* Add conda config to install new customer-managed environments under home directory by default
* Add supervisord to manage the Jupyter server process

To start the docker container with Jupyter server running:

```
docker run -p 8888:8888 <image-name> --entrypoint entrypoint-jupyter-server
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

balajisankar15 commented:
1. Can you rebase to pick the latest changes ? I synced 1.0.0-beta to bring in the latest commits from the main branch
2. Can we just limit this pr to include just changes to the template, src/main.py and README changes. Once this is merged, you can send a follow up PR to merge the updates to 1.0.0-beta

zuoyuanh commented:
Thanks for the callout -

1. I rebased the PR to pick up the latest changes
2. Yes, I refactored the PR and now it includes template and README changes
